virtual-scroller-element.js

```diff
@@ -32,9 +36,12 @@ export class VirtualScrollerElement extends HTMLElement {
       }
       return childTemplate.cloneNode(true);
     };
-    this[_updateElement] = (element, index) => element.textContent = index;
+    this[_updateElement] = (element, item) => element.textContent =
+        item.toString();
     this[_recycleElement] = (element) => this[_nodePool].push(element);
-    this[_elementKey] = null;
+
+    this[_itemSource] = this[_rawItemSource] = null;
```

Comment:
What is the point of _rawItemSource? Just to prevent re-rendering if you set the same object? I think it's OK in that case to re-render.

Reply:
Yes, it's avoiding rerendering and invalidation of the [internal key/Element map](https://github.com/Polymer/lit-element#minimal-example).
ItemSource determines the `elementKey` function, which when gets changed causes invalidation of the internal map.

This would be a problem in patterns like LitElement where we always set properties/attributes when rendering/updating the dom - see https://github.com/Polymer/lit-element#minimal-example
As an example, imagine I'm rendering virtual-scroller and setting the contacts **array**
```js
_render({ contacts }) { // typeof contacts === 'Array' 
  return html`<virtual-scroller itemSource="${contacts}"></virtual-list>`;
}
```

Reply:
Oh, wow, so how does LitElement handle things like `imgElement.src`, where setting it causes a whole new fetch?

Reply:
it does check for identity internally before actually updating the property/attribute.
In our case, the getter always returns an ItemSource object, whereas the setter receives an Array.

Reply:
Ah, interesting point. Happy to leave in the check for now and think harder later. Maybe we should reconsider that automatic conversion? Or maybe this is just a special case though and keeping the check is OK.

Reply:
I think it would be better to return the rawItemSource in the getter and use the actual ItemSource internally. Like that, the user will get what he's set. We'll still need to do the identity check to know if to update the internal ItemSource tho

Reply:
I'll open an issue on this, so we can go ahead and merge here 👌

Reply:
Filed https://github.com/valdrinkoshi/virtual-scroller/issues/93 - feel free to merge this!